File: checkifsorted.py
import sys


def is_sorted(arr):
    # Verifies non-decreasing order across the whole array
    return all(arr[i] <= arr[i + 1] for i in range(len(arr) - 1))


def main():
    data = list(map(int, sys.stdin.read().strip().split()))
    if not data:
        return
    n, *rest = data
    arr = rest[:n]
    if is_sorted(arr):
        print("the array is sorted")
    else:
        print("the array is not sorted")


if __name__ == "__main__":
    main()

File: placementprepration/checkifsorted.cpp
#include<iostream>
#include<algorithm>
using namespace std;
bool sorted(int arr[],int n){
    for(int i=0;i<n;i++){
        if(arr[i]>arr[i+1]){
            return false;
        }
        else{
            return true;
        }
    }
}
int main(){
    int n;
    cin>>n;
    int arr[n];
    for(int i=0;i<n;i++){
        cin>>arr[i];
    }
    if(sorted(arr,n)){
        cout<<"the array is sorted";
    }
    else{
        cout<<"the array is not sorted";
    }
    return 0;
}